### PR TITLE
One letter codes

### PR DESCRIPTION
**Pull request by cxhernandez**
As discussed in https://github.com/mdtraj/mdtraj/issues/870

### CODE REVIEW

**Comment by cxhernandez** [comment]
I decided to include all the amino acids recognized by the PDB Chemical Component Dictionary, but this can easily be changed if you all decide that's overkill. 

I also added a `to_fasta` method to the `Topology` class.

**Comment by cxhernandez** [comment]
Also, need to add a build test still

**Comment by rmcgibbo** [comment]
Where exactly do these codes come from? I haven't looked at the PBD component dictionary before.

**Comment by cxhernandez** [comment]
here: ftp://ftp.wwpdb.org/pub/pdb/data/monomers/components.cif

**Comment by swails** [comment]
That's the CIF file.  Here's the page for it: http://www.wwpdb.org/data/ccd

**Comment by swails** [comment]
There is a field `one_letter_code` that has this information.

**Comment by rmcgibbo** [comment]
Alright. I think I'm the least knowledgeable person in the room here about most of these issues, I'm probably not the best person to do code review.

I'll let you guys decide what makes sense for this feature (full list of all amino acids in the dictionary, or some truncated set), do code review, and then merge.

**Comment by cxhernandez** [comment]
Any thoughts @swails, @kyleabeauchamp, or anybody else?

File: mdtraj/core/topology.py [hunk 1]
```diff
@@ -196,7 +199,9 @@ def __repr__(self):
         return "<%s at 0x%02x>" % (self._string_summary_basic(), id(self))
 
     def _string_summary_basic(self):
-        return "mdtraj.Topology with %d chains, %d residues, %d atoms, %d bonds" % (self.n_chains, self.n_residues, self.n_atoms, len(self._bonds))
+        return "mdtraj.Topology with %d chains, %d residues, \
+                %d atoms, %d bonds" % (self.n_chains, self.n_residues,
+                                       self.n_atoms, len(self._bonds))
```
Comment (swails) [inline mdtraj/core/topology.py]: This should be changed.  All that gets cancelled is the `\n` -- the indentation spaces still get added to the returned string.  Change to this:

```Python
    def _string_summary_basic(self):
        return ("mdtraj.Topology with %d chains, %d residues, "
                "%d atoms, %d bonds" % (self.n_chains, self.n_residues,
                                        self.n_atoms, len(self._bonds)))

File: mdtraj/core/topology.py [hunk 2]
```diff
@@ -297,7 +327,8 @@ def to_openmm(self, traj=None):
             angles = traj.unitcell_angles[0]
 
             if np.linalg.norm(angles - 90.0) > 1E-4:
-                raise(ValueError("Unitcell angles must be 90.0 to use in OpenMM topology."))
+                raise(ValueError("Unitcell angles must be 90.0 to use \
+                                 in OpenMM topology."))
```
Comment (swails) [inline mdtraj/core/topology.py]: Same here.  When breaking strings across lines, use the technique I described above.  This adds unnecessary spaces.  Also, this is no longer true (OpenMM supports triclinics), but I'll do this in a different PR.

File: mdtraj/core/residue_names.py [hunk 3]
```diff
@@ -20,35 +20,849 @@
...
+                     'XX1': 'K',
+                     'YCM': 'C',
+                     'YOF': 'Y',
+                     'YTH': 'T',
+                     'Z01': 'A',
+                     'ZAL': 'A',
+                     'ZCL': 'F',
+                     'ZFB': 'X',
+                     'ZU0': 'T',
+                     'ZZJ': 'A'}
```
Comment (swails) [inline mdtraj/core/residue_names.py]: Are all of these codes derived amino acids?  I don't think they are.  Single-letter codes should be separated from amino acid identification, IMO

Comment (swails) [inline mdtraj/core/residue_names.py]: (Unless I'm wrong and these are all amino acids or modified versions)

**Comment by swails** [comment]
I added my reviews.  I'm +1 on this.

File: mdtraj/core/topology.py (hunk 1 above)
Comment (cxhernandez) [inline mdtraj/core/topology.py]: Ah, good point.

File: mdtraj/core/residue_names.py (hunk 3 above)
Comment (cxhernandez) [inline mdtraj/core/residue_names.py]: In curating this `dict`, I grep'd for components that had a [`_chem_comp.type`](http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx.dic/Items/_chem_comp.type.html) listed as 'peptide-like' or 'peptide-linking' in the `CIF`.

I checked a few of the weird ones, and they turned out to be modified amino acids. I'm assuming the bulk of these are modified residues but could definitely be wrong.

Comment (swails) [inline mdtraj/core/residue_names.py]: I'm satisfied. That's exactly what I would have done.

**Comment by mpharrigan** [comment]
Add to selection language and docs for selection lang?

**Comment by cxhernandez** [comment]
How painful would that be? I've never touched that part of the code.

**Comment by mpharrigan** [comment]
Should be a one line change addition? Let me know if you want help

**Comment by cxhernandez** [comment]
Oh, okay cool! 

Also, any idea why this build keeps failing? I thought I started from master....

**Comment by kyleabeauchamp** [comment]
No complaints on my end

**Comment by kyleabeauchamp** [comment]
No idea about the travis failures, looks like a real test failure, but I'm not sure where

**Comment by cxhernandez** [comment]
`ACE` and `NME` weren't included in the PDB CCD for some reason. I think that screwed up some of the atom selection tests.

**Comment by cxhernandez** [comment]
okay just gotta add some docs + tests now

**Comment by cxhernandez** [comment]
Do I need to update anymore of the docs?

File: mdtraj/core/residue_names.py [hunk 4]
```diff
@@ -20,35 +20,851 @@
...
+    'SB', 'SCN', 'SE4', 'SEK', 'SM', 'SMO', 'SO3', 'SO4', 'SOD', 'SR',
+    'Sm', 'Sn', 'T1A', 'TB', 'TBA', 'TCN', 'TEA', 'THE', 'TL', 'TMA',
+    'TRA', 'UNX', 'V', 'V2+', 'VN3', 'VO4', 'W', 'WO5', 'Y1', 'YB', 'YB2',
+    'YH', 'YT3', 'ZN', 'ZN2', 'ZN3', 'ZNA', 'ZNO', 'ZO3'])
 
 _WATER_RESIDUES = frozenset([
     'H2O', 'HHO', 'OHH', 'HOH', 'OH2', 'SOL',
     'WAT', 'TIP', 'TIP2', 'TIP3', 'TIP4'])
+
+_AMINO_ACID_CODES = {'ACE': None,
```
Comment (mpharrigan) [inline mdtraj/core/residue_names.py]: Can we condense this to fewer lines?

Comment (cxhernandez) [inline mdtraj/core/residue_names.py]: fineeeee

**Pull request merged**